addon/components/cf-content.js

```diff
@@ -170,9 +183,124 @@ export default Component.extend(ComponentQueryManager, {
-      } finally {
-        window.scrollTo(0, 0);
       }
     }
-  )
+  ),
+
+  _sections: filterBy(
+    "rootDocument.visibleFields",
+    "visibleInNavigation",
+    true
+  ),
+
+  _currentSection: computed("_sections.[]", "section", function() {
+    return this._sections.find(s => s.question.slug === this.section);
+  }),
+
+  _currentSubSection: computed(
+    "_currentSubSections.[]",
+    "subSection",
+    function() {
+      return this._currentSubSections.find(
+        s => s.question.slug === this.subSection
+      );
+    }
+  ),
+
+  _currentSubSections: filterBy(
+    "_currentSection.childDocument.visibleFields",
+    "visibleInNavigation",
+    true
+  ),
+
+  _currentSectionIndex: computed("_sections.[]", "section", function() {
+    return this._sections.indexOf(this._currentSection);
+  }),
+
+  _previousSection: computed("_currentSectionIndex", function() {
+    return this._currentSectionIndex > 0
+      ? this._sections[this._currentSectionIndex - 1]
+      : null;
+  }),
+
+  _nextSection: computed(
+    "_currentSectionIndex",
+    "_sections.length",
+    function() {
+      return this._currentSectionIndex < this._sections.length
+        ? this._sections[this._currentSectionIndex + 1]
+        : null;
+    }
+  ),
+
+  _previousSubSections: filterBy(
+    "_previousSection.childDocument.visibleFields",
+    "visibleInNavigation",
+    true
+  ),
+
+  _nextSubSections: filterBy(
+    "_nextSection.childDocument.visibleFields",
+    "visibleInNavigation",
+    true
+  ),
+
+  adjacentSections: computed(
+    "_previousSubSections.[]",
+    "_currentSubSections.[]",
+    "_nextSubSections.[]",
+    function() {
+      return [
+        ...buildParams(this._previousSection, this._previousSubSections),
+        ...buildParams(this._currentSection, this._currentSubSections),
+        ...buildParams(this._nextSection, this._nextSubSections)
+      ];
+    }
+  ),
+
+  sectionIndex: computed(
+    "adjacentSections.[]",
+    "section",
+    "subSection",
+    function() {
+      return this.adjacentSections.findIndex(
+        s => s.section === this.section && s.subSection === this.subSection
+      );
+    }
+  ),
+
+  previousSection: computed("adjacentSections.[]", "sectionIndex", function() {
+    return this.sectionIndex > 0
+      ? this.adjacentSections[this.sectionIndex - 1]
+      : null;
+  }),
+
+  nextSection: computed("adjacentSections.[]", "sectionIndex", function() {
+    return this.sectionIndex < this.adjacentSections.length
+      ? this.adjacentSections[this.sectionIndex + 1]
+      : null;
+  }),
+
+  // eslint-disable-next-line ember/no-observers
+  _displayedDocumentChanged: observer(
+    "displayedDocument",
+    "nextSection",
+    function() {
+      this.displayedDocumentChanged.perform();
+    }
+  ),
+
+  displayedDocumentChanged: task(function*() {
+    if (isDisplayableDocument(this.displayedDocument) || !this.nextSection) {
+      return;
+    }
+
+    yield timeout(100);
```

Comment:
Why not just the next runloop?

Reply:
Because I want it to run just once and `Ember.debouce` somehow didn't work..

Reply:
I gotta try another solution here. This is not as clean as we want it to be..

Reply:
AFAIK yielding a timeout is the 'official' way to debounce inside a task.

Reply:
If this is "best practice" I'd also be ok with it. Just the number `100` seem artificial.

Reply:
Here's the debounce example from the official docs:
http://ember-concurrency.com/docs/examples/autocomplete/

And aren't the debounce values (almost) always arbitrary? I personally use 300ms but that's a value I got by trial and error while developing frontend widgets.

Reply:
> a timeout is the 'official' way to debounce inside a task.

`yield timeout(x)` is definitely best practice for debouncing.

> And aren't the debounce values (almost) always arbitrary? I personally use 300ms but that's a value I got by trial and error while developing frontend widgets.

Problem is, this debounce is not a UX thing here. What I was trying to do is waiting for all the `hiddenTask`s to finish. However I have no way of knowing when they are started or finished.

I'll look more into this, debouncing does not seem like the right way to do this..

Reply:
Done!